packages/core/src/api/positionMapping.ts

```diff
@@ -0,0 +1,126 @@
+import { Mapping } from "prosemirror-transform";
+import {
+  absolutePositionToRelativePosition,
+  relativePositionToAbsolutePosition,
+  ySyncPluginKey,
+} from "y-prosemirror";
+import type { BlockNoteEditor } from "../editor/BlockNoteEditor.js";
+import * as Y from "yjs";
+import type { ProsemirrorBinding } from "y-prosemirror";
+import type {
+  DefaultInlineContentSchema,
+  DefaultStyleSchema,
+} from "../blocks/defaultBlocks.js";
+import type { DefaultBlockSchema } from "../blocks/defaultBlocks.js";
+import type {
+  BlockSchema,
+  InlineContentSchema,
+  StyleSchema,
+} from "../schema/index.js";
+
+/**
+ * This is used to track a mapping for each editor. The mapping stores the mappings for each transaction since the first transaction that was tracked.
+ */
+const editorToMapping = new Map<BlockNoteEditor<any, any, any>, Mapping>();
+
+/**
+ * This initializes a single mapping for an editor instance.
+ */
+function getMapping(editor: BlockNoteEditor<any, any, any>) {
+  if (editorToMapping.has(editor)) {
+    // Mapping already initialized, so we don't need to do anything
+    return editorToMapping.get(editor)!;
+  }
+  const mapping = new Mapping();
+  editor._tiptapEditor.on("transaction", ({ transaction }) => {
+    mapping.appendMapping(transaction.mapping);
+  });
+  editor._tiptapEditor.on("destroy", () => {
+    // Cleanup the mapping when the editor is destroyed
+    editorToMapping.delete(editor);
+  });
+
+  // There only is one mapping per editor, so we can just set it
+  editorToMapping.set(editor, mapping);
+
+  return mapping;
+}
+
+/**
+ * This is used to keep track of positions of elements in the editor.
+ * It is needed because y-prosemirror's sync plugin can disrupt normal prosemirror position mapping.
+ *
+ * It is specifically made to be able to be used whether the editor is being used in a collaboratively, or single user, providing the same API.
+ *
+ * @param editor The editor to track the position of.
+ * @param position The position to track.
+ * @param side The side of the position to track. "left" is the default. "right" would move with the change if the change is in the right direction.
+ * @returns A function that returns the position of the element.
+ */
+export function trackPosition<
+  BSchema extends BlockSchema = DefaultBlockSchema,
```

Comment:
generic can be removed, editor can just be <any,any,any> as the `trackPosition` code doesn't care about the schema